src/appkit/window/controller/class.rs

```diff
@@ -4,23 +4,23 @@
 use std::sync::Once;
 
 use objc::class;
-use objc::declare::ClassDecl;
+use objc::declare::ClassBuilder;
 use objc::runtime::Class;
 
 use crate::appkit::window::{WindowDelegate, WINDOW_DELEGATE_PTR};
 
 /// Injects an `NSWindowController` subclass, with some callback and pointer ivars for what we
 /// need to do.
-pub(crate) fn register_window_controller_class<T: WindowDelegate>() -> *const Class {
-    static mut DELEGATE_CLASS: *const Class = 0 as *const Class;
+pub(crate) fn register_window_controller_class<T: WindowDelegate>() -> &'static Class {
+    static mut DELEGATE_CLASS: Option<&'static Class> = None;
```

Comment:
Does this need to be an `Option<&'static Class>` because it was initialized as `0 as *const Class` before? I think the only part that feels odd to me is the `unsafe { DELEGATE_CLASS.unwrap() }`.

Reply:
The static is created before the program begins, so it has to be initialized to some value, in this case `None`. When the function is called the first time, the class is created and the static is set to the new value.
The unsafety in `unsafe { DELEGATE_CLASS.unwrap() }` comes from the fact that `DELEGATE_CLASS` is a mutable static, and hence can only be accessed behind `unsafe`. Data races are prevented by `Once`.

In essence, what we do here with `DELEGATE_CLASS` and `Once` is similar to [`once_cell::sync::Lazy`](https://docs.rs/once_cell/latest/once_cell/sync/struct.Lazy.html) and the nightly [`std::cell::LazyCell`](https://doc.rust-lang.org/nightly/std/cell/struct.LazyCell.html).